provisioning/hare_setup.py

```diff
@@ -0,0 +1,305 @@
+        try:
+            if not is_cluster_running() and bootstrap_cluster() != 0:
+                logging.error('Failed to bootstrap the custer')
+                exit(1)
+
+            while is_cluster_running():
+                shutdown_cluster()
+
+        except Exception as error:
+            logging.error('Error while initializing the cluster (%s)',
+                          error)
+            exit(-1)
+
+class Test(argparse.Action):
+    """
+    Read CDF file and compare fields with output of 'hctl status'
+    Will start the cluster(and shutdown before exiting) if not already started
+    """
+    def __call__(self, parser, namespace, values, option_string=None):
+        try:
+            if not is_cluster_running() and bootstrap_cluster() != 0:
+                logging.error('Failed to bootstrap the cluster')
+                exit(-1)
+
+            cluster_status = check_cluster_status()
+
+            while is_cluster_running():
+                shutdown_cluster()
+
+            if cluster_status:
+                logging.error('Cluster status reports failure')
+                exit(-1)
+
+        except Exception as error:
+            logging.error('Error while checking cluster status (%s)',
+                          error)
+            exit(-1)
+
+class SupportBundle(argparse.Action):
+    def __call__(self, parser, namespace, values, option_string=None):
+        try:
+            # Default target directory is /tmp/hare
+            if os.system('hctl reportbug') != 0:
+                logging.error('Failed to generate support bundle')
+                exit(-1)
+        except Exception as error:
+            logging.error('Error while generating support bundle (%s)',
+                          error)
+            exit(-1)
+
+class Cleanup(argparse.Action):
+    def __call__(self, parser, namespace, values, option_string=None):
+        try:
+            exit(0)
+        except Exception as error:
+            logging.error('Failed to perform cleanup')
+            exit(-1)
+
+def checkRpm(rpm_name):
+    rpm_list = subprocess.Popen(["rpm", "-qa"],
+                                stdin=subprocess.PIPE,
+                                stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE,
+                                env={},
+                                encoding='utf8')
+    rpm_search = subprocess.Popen(["grep", "-q", rpm_name],
+                                  stdin=rpm_list.stdout,
+                                  stdout=subprocess.PIPE,
+                                  stderr=subprocess.PIPE,
+                                  env={},
+                                  encoding='utf8')
+    out, err = rpm_list.communicate()
+    rpm_search.wait()
+    logging.debug("Output: {}".format(out))
+    logging.debug("stderr: {}".format(err))
+
+    return rpm_search.returncode
+
+
+def is_cluster_running() -> bool:
+    if os.system('hctl status >/dev/null') == 0:
```

Comment:
```
if <bool>:
   return <bool>
else
   return <bool>
```

looks really weird. Please consider replacing it with something like 
```python
return os.system('hctl status >/dev/null') == 0
```

Reply:
Done